src/app/chip-zcl-zpro/command-encoder/decoder.c

```diff
@@ -103,18 +103,10 @@ void printApsFrame(EmberApsFrame * frame)
            frame->groupId, frame->sequence, frame->radius);
 }
 
-void printCommandInfo(void * buffer)
+uint16_t extractMessage(uint8_t * buffer, uint16_t buffer_length, void ** msg)
 {
-    void * msg = NULL;
-    extractMessage(buffer, &msg);
-    uint8_t val = (uint8_t) * ((uint8_t *) (msg + 2));
-    printf("Encoded command id %d\n", val);
-}
-
-uint32_t extractMessage(void * buffer, void ** msg)
-{
-    uint32_t result = 0;
-    if (msg)
+    uint16_t result = 0;
+    if (msg && buffer_length > 13)
```

Comment:
If msg is non-null but the buffer is too short, we may want to still set `*msg` to some valid pointer (e.g. `buffer`?) and claim 0 length...  Or maybe set `*msg` to null?

Reply:
Done. I set *msg to null